test/filewatch_test.erl

```diff
@@ -0,0 +1,89 @@
+-module(filewatch_test).
+-include_lib("eunit/include/eunit.hrl").
+
+-define(
+   rapid_events_test(Files, NEvents),
+   Dir = create_sandbox(?FUNCTION_NAME),
+   Pairs = [{Dir ++ File, File} || File <- Files],
+   {ok, P} = filewatch:start(self(), Pairs),
+   [start_sender(Path, NEvents) || {Path, _} <- Pairs],
+   State = collect_events(Files),
+   filewatch:stop(P),
+   rm(Dir),
+   ?assert(State)
+  ).
+
+one_file_few_rapid_events_test() ->
```

Comment:
I would recommend adding a simple one or two event tests to ensure that we get event notifications for creating replacing and dleting a file.